File: python/queue.py
class Queue():

    def __init__(self):
        self.items = []
    
    def Enqueue(self, data):
        self.items.append(data)

    def Dequeue(self):
        self.items.pop(0)

    def isEmpty(self):
        if self.items == []:
            return True
        else:
            return False

q1 = Queue()
for i in range(10):
    q1.Enqueue(i)
print(q1.items)
q1.Dequeue()
q1.Dequeue()
print(q1.items)
q1.Enqueue(10)
print(q1.items)
q1.Dequeue()
print(q1.items)

File: go/queue.go
package main

import "fmt"

func main() {
	var queue Queue
	for i := 1; i < 9; i++ {
		queue.Enqueue(i)
	}
	fmt.Println(queue)
	fmt.Println(queue.Len())
	queue.Dequeue()
	queue.Dequeue()
	fmt.Println(queue)

}

type Queue []interface{}

func (q *Queue) Enqueue(val interface{}) interface{} {
	*q = append(*q, val)
	return val
}

func (q *Queue) Dequeue() interface{} {
	*q = (*q)[1:]
	return (*q)[0]
}

func (q *Queue) isEmpty() bool {
	if len(*q) == 0 {
		return true
	}
	return false
}
func (q *Queue) Len() int {
	return len(*q)
}
